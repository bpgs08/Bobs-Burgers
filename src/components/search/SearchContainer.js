import React, { useEffect } from "react";
import { connect } from "react-redux";
import Search from "./Search";

import { addArticlesAction } from "../../actions/addArticlesAction";

let SearchContainer = ({
  maxPages,
  searchParam,
  currentPage,
  addArticlesAction,
}) => {
  let mountArticles = () => {
    return addArticlesAction({
      searchParam: "",
      currentPage: currentPage,
    });
  };

  const useMountEffect = (func) => useEffect(func, []);

  useMountEffect(mountArticles);

  let handleSearchParam = (e) => {
    addArticlesAction({
      searchParam: e.target.value,
      currentPage: currentPage,
    });
  };

  let handlePageChange = (e) => {
    let numChange = parseInt(e.target.value, 10);
    addArticlesAction({
      searchParam,
      currentPage: currentPage + numChange,
    });
  };

  return (
    <Search
      maxPages={maxPages}
      searchParam={searchParam}
      currentPage={currentPage}
      handleSearchParam={handleSearchParam}
      handlePageChange={handlePageChange}
    />
  );
};

const mapDispatchToProps = {
  addArticlesAction,
};

const mapStateToProps = ({ addArticlesReducer }) => {
  return {
    maxPages: addArticlesReducer.maxPages,
    searchParam: addArticlesReducer.searchParam,
    currentPage: addArticlesReducer.currentPage,
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(SearchContainer);
