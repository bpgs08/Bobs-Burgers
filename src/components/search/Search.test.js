import React from "react";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Provider } from "react-redux";
import store from "../../store";
import renderer from "react-test-renderer";
import SearchContainer from "./SearchContainer";
Enzyme.configure({ adapter: new Adapter() });

describe("<SearchContainer /> component", () => {
  it("snapshot", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <SearchContainer />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
