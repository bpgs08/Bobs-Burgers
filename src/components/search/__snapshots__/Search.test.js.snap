// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchContainer /> component snapshot 1`] = `
<div
  className="sc-AxiKw ckaBAP"
>
  <input
    className="sc-AxjAm btDDba"
    id="SearchHere"
    onChange={[Function]}
    placeholder="Search here"
    type="text"
  />
</div>
`;
